src/services/auth/service.ts

```diff
@@ -45,37 +45,28 @@ export class AuthService {
     destination.searchParams.set('url', encodeURIComponent(redirectionUrl));
     const link = destination.toString();
 
-    const lang = member.lang;
-
-    const translated = this.mailerService.translate(lang);
-    const subject = translated(MAIL.SIGN_UP_TITLE);
-    const greetingsAndSignupText = `${translated(MAIL.GREETINGS)} ${translated(MAIL.SIGN_UP_TEXT)}`;
-    const html = `
-    ${this.mailerService.buildText(greetingsAndSignupText)}
-    ${this.mailerService.buildButton(link, translated(MAIL.SIGN_UP_BUTTON_TEXT))}
-    ${this.mailerService.buildText(
-      translated(MAIL.USER_AGREEMENTS_MAIL_TEXT, {
-        signUpButtonText: translated(MAIL.SIGN_UP_BUTTON_TEXT),
-        graaspLandingPageOrigin: GRAASP_LANDING_PAGE_ORIGIN,
-      }),
-      // Add margin top of -15px to remove 15px margin bottom of the button.
-      { 'text-align': 'center', 'font-size': '10px', 'margin-top': '-15px' },
-    )}
-    ${this.mailerService.buildText(translated(MAIL.SIGN_UP_NOT_REQUESTED))}`;
-
-    const footer = this.mailerService.buildFooter(lang);
+    const mail = new MailBuilder({
+      subject: { text: MAIL.SIGN_UP_TITLE },
+      lang: member.lang,
+    })
+      .addText(MAIL.GREETINGS)
+      .addText(MAIL.SIGN_UP_TEXT)
+      .addButton(MAIL.SIGN_UP_BUTTON_TEXT, link)
+      .addUserAgreement(MAIL.SIGN_UP_BUTTON_TEXT)
```

Comment:
It looks strange to me that we are providing text in this function, and the text is the same as the signup button ? either rename the translation string to better fit the purpose, or remove the argument and have the text be defined inside the builder function.

Reply:
I have changed the actual string for the user agreement so that it no longer requires the button text. This way I could also remove the parameter for the `addUserAgreement` function.